File: frontend/src/types/role.ts
/**
 * 角色实体
 */
export interface Role {
    id: number
    code: string
    name: string
    description?: string
    sort: number
    enabled: boolean
    createTime: string
    updateTime: string
}

/**
 * 角色创建请求
 */
export interface RoleCreateRequest {
    code: string
    name: string
    description?: string
    sort?: number
    enabled?: boolean
}

/**
 * 角色更新请求
 */
export interface RoleUpdateRequest {
    name?: string
    description?: string
    sort?: number
    enabled?: boolean
}

/**
 * 角色查询参数
 */
export interface RoleQueryParams {
    pageNumber?: number
    pageSize?: number
    name?: string
    code?: string
    enabled?: boolean
}

File: backend/src/main/java/com/antigravity/module/role/Role.java
package com.antigravity.module.role;

import com.antigravity.common.BaseEntity;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serial;

/**
 * 角色实体
 *
 * @author Antigravity Team
 * @since 1.0.0
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_role")
public class Role extends BaseEntity {

    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 角色编码（唯一）
     */
    private String code;

    /**
     * 角色名称
     */
    private String name;

    /**
     * 角色描述
     */
    private String description;

    /**
     * 排序号
     */
    private Integer sort;

    /**
     * 是否启用
     */
    private Boolean enabled;

    /**
     * 是否删除
     */
    @TableLogic
    private Boolean isDeleted;

}
